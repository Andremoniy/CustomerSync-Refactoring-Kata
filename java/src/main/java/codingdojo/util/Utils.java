package codingdojo.util;

import codingdojo.misc.Customer;
import codingdojo.misc.CustomerDataLayer;
import codingdojo.match.CustomerMatches;
import codingdojo.misc.ShoppingList;

public class Utils {
    public static void subLoadCompanyCustomer(CustomerMatches matches, Customer matchByExternalId, Customer matchByMasterId, Customer matchByCompanyNumber) {
        if (matchByExternalId != null) {
            matches.setCustomer(matchByExternalId);
            matches.setMatchTerm("ExternalId");
            if (matchByMasterId != null) matches.addDuplicate(matchByMasterId);
        } else {
            if (matchByCompanyNumber != null) {
                matches.setCustomer(matchByCompanyNumber);
                matches.setMatchTerm("CompanyNumber");
            }
        }
    }

    public static void updateShoppingList(Customer customer, ShoppingList consumerShoppingList, CustomerDataLayer customerDataLayer1) {
        customer.addShoppingList(consumerShoppingList);
        customerDataLayer1.updateShoppingList(consumerShoppingList);
        customerDataLayer1.upd(customer);
    }
}
