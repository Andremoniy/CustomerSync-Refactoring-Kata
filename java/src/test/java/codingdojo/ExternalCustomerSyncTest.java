package codingdojo;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertFalse;


public class ExternalCustomerSyncTest {

    @Test
    public void testSyncByExternalId(){
        String externalId = "12345";

        ExternalCustomer externalCustomer = createExternalCustomer();
        externalCustomer.setExternalId(externalId);

        Customer customer = new Customer();
        customer.setExternalId(externalId);
        customer.setCompanyNumber(externalCustomer.getCompanyNumber());
        customer.setCustomerType(CustomerType.COMPANY);
        customer.setInternalId("45435");

        FakeDatabase db = new FakeDatabase();
        db.addCustomer(customer);
        CustomerSync sut = new CustomerSync(db);

        StringBuilder toAssert = new StringBuilder();
        toAssert.append("BEFORE:\n");
        toAssert.append(db.printContents());

        toAssert.append("\nSYNCING THIS:\n");
        toAssert.append(ConsumerPrinter.print(externalCustomer, ""));

        // ACT
        boolean created = sut.syncWithDataLayer(externalCustomer);

        assertFalse(created);
        toAssert.append("\nAFTER:\n");
        toAssert.append(db.printContents());
        Approvals.verify(toAssert);
    }

    private ExternalCustomer createExternalCustomer() {
        ExternalCustomer externalCustomer = new ExternalCustomer();
        externalCustomer.setExternalId("12345");
        externalCustomer.setName("Standard External Customer");
        externalCustomer.setAddress(new Address("123 main st", "Helsingborg", "SE-123 45"));
        externalCustomer.setCompanyNumber("470813-8895");
        externalCustomer.setPreferredStore("Nordstan");
        externalCustomer.setShoppingLists(Arrays.asList(new ShoppingList("lipstick", "blusher")));
        return externalCustomer;
    }
}
