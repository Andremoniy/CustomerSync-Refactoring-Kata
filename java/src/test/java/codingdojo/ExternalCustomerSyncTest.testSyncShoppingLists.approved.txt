BEFORE:
Fake Database.
All Customers {

    Customer {
        externalId='12345'
        masterExternalId='null'
        companyNumber='470813-8895'
        internalId='45435'
        name='null'
        customerType=COMPANY
        preferredStore='null'
        address='null'
        shoppingLists=[[eyeliner, blusher]]
    }

}
All Shopping Lists
[[eyeliner, blusher]]
SYNCING THIS:
ExternalCustomer {
    externalId='12345'
    companyNumber='470813-8895'
    name='Standard External Customer'
    preferredStore='Nordstan'
    address='123 main st, SE-123 45 Helsingborg'
    shoppingLists=[[lipstick, blusher]]
}
AFTER:
Fake Database.
All Customers {

    Customer {
        externalId='12345'
        masterExternalId='null'
        companyNumber='470813-8895'
        internalId='45435'
        name='Standard External Customer'
        customerType=COMPANY
        preferredStore='Nordstan'
        address='123 main st, SE-123 45 Helsingborg'
        shoppingLists=[
            [eyeliner, blusher]
            [lipstick, blusher]
        ]
    }

}
All Shopping Lists
[
    [eyeliner, blusher]
    [lipstick, blusher]
]